mpeg2/src/temi.rs

```diff
@@ -0,0 +1,282 @@
+use crate::{DecodeError, EncodeError};
+use byteorder::{BigEndian, ByteOrder};
+use core2::io::Write;
+
+pub const AF_DESCR_TAG_TIMELINE: u8 = 0x04;
+
+#[derive(Debug, PartialEq, PartialOrd, Copy, Clone)]
+#[allow(non_camel_case_types)]
+#[repr(C)]
+pub enum TimestampLength {
+    None = 0,
+    ThirtyTwoBits = 1,
+    SixtyFourBits = 2,
+}
+
+impl Default for TimestampLength {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct TEMITimelineDescriptor {
+    pub has_timestamp: TimestampLength,
+    pub timescale: u32,
+    pub media_timestamp: u64,
+
+    pub ntp_timestamp: Option<u64>,
+    pub ptp_timestamp: Option<u128>,
+
+    pub drop: bool,
+    pub frames_per_tc_seconds: u16,
+    pub duration: u16,
+    pub has_timecode: TimestampLength,
```

Comment:
You should make a separate enum type for this, since the timecode is either 24 or 64 bits (not 32 or 64 like the timestamp).

```suggestion
    pub has_timecode: TimecodeLength,
```

Reply:
`TimestampLength` enum is used by both `has_timecode` and `has_timestamp`, maybe rename this enum to `TimeLength`?

Reply:
My point is that it shouldn't be used for both of them. Because the timecode is either 24 bits or 64 bits, but the timestamp is either 32 or 64 bits. They don't have the same possible lengths.

Reply:
I see you point, maybe we can define the `enum` as follows,
```rust
pub enum TimeFieldLength {
    None = 0,
    Short = 1,
    Long = 2,
}
```
Then it makes sense to be used by both of them. `Short` means length is less than or equal to 32 bits.

Reply:
🤷‍♂️ Either works.